### DISCUSSION TITLE
Juggluco e SmartBand

### DISCUSSION THREAD

**Discussion by SideTipeOne** [j-kaltes/Juggluco#198] [Ideas]
**Hi everyone**.

Thanking you again for the project of this App, I would like to ask you for some info regarding the use of the App with systems such as SmartWatches/**SmartBand** etc.

I saw that for the Juggluco App there is also a version available that can be installed on certain smartwatches, which would somehow work independently with the Sensor being able to scan it (for compatible models equipped with NFC).
I wanted to know if there is a way to make everything work on SmartBand too?
These are products that from what I understand work slightly differently compared to a SmartWatch that has its own operating system, and I don't know what potential technical limits there may be.
From some info taken online, and interpreted with a lot of technical unpreparedness, I saw that to obtain the reading of the values ​​on SmartBand you rely on a "bridge system". Composed by the communication of 2 Apps, Juggluco and Xdrip in general (although for the latter there would also be others that would perform its same role in this "reading system"), which would somehow communicate via Bluetooth and data connection with the SmartBand allowing the viewing of a dial with the data.
Basically I seem to have understood that Juggluco (which reads the sensor) from the mobile phone, sends the sensor reading data to the server of this App and that then the other App, "Xdrip" (or others that I seem to have understood work a bit in the same way) also installed on the mobile phone, retrieves via data connection (internet) from this server the reading data that Juggluco sends, making them then visible and potentially available for viewing a dial on a SmartBand.
From what I understand, however, this system has the defect of having a notable "LAG", and therefore the values ​​​​displayable through this system are at least 10 minutes old, and also potentially at risk if you were to find yourself in areas with little, or unstable, internet coverage.

I therefore wanted to know if it is possible to eliminate this "necessary" bridge, and have Juggluco send the data directly via Bluetooth to a SmartBand with a dial.
I don't know if something like this is possible and if a similar feature is being developed, or a second app for example to be installed on the mobile phone, which could work as a simultaneous reader of sensor data or Juggluco main data (sensor reader) that can make the reading of that data received available to a SmartBand via Bluetooth (therefore using a Bluetooth output connection of another app dedicated only to this) being able to eliminate the bridge that requires the internet connection (as in the case I explained before with Xdrip and server) and therefore being able to have a more secure connection system, in real time and without potential "LAGs".

I don't have the expertise in the matter to realistically imagine the creation of something similar, or if the simultaneous Bluetooth connection involved could cause problems.

But I would like to understand it, and understand what to expect in this regard.

Thank you very much for your patience.

**Comment by j-kaltes** [comment]
Juggluco for WearOS doesn’t use NFC to scan the sensor. It can receive the glucose values via Bluetooth after the sensor is scanned by a smartphone running Android connected to the watch.
If you look in [Left menu→Watch→Help](https://www.juggluco.nl/Jugglucohelp/watchinfo.html) , you see that are a lot of other watches that can receive glucose values from Juggluco on a phone that is connected to the sensor. Not all of them show a considerable lag.
A lot of smartbands can be connected to Watchdrip or Gadgetbridge that receive glucose values from Juggluco. They display the received glucose value a few seconds after Juggluco receives the glucose value. 
The disadvantage of Watchdrip is that is takes 10 second to update the watch face. This wouldn’t be such a large problem, where it not that during this time the watch displays “_synchronizing_” instead of the previous watch face, thus making the watch one sixth of the time unusable. 
The disadvantage of Gadgetbridge it that you need a special watch face that displays the weather information as glucose values, because Juggluco sends glucose values as weather information to gadgetbridge: https://github.com/j-kaltes/Juggluco/discussions/33

**Reply by SideTipeOne** [reply]
Thank you for this valuable information and for your reply.
I want to apologize in advance for my profound and potential ignorance on the subject, and if I can cause confusion 😇.
Thanks in advance for your patience and your time🙏🏻🙏🏻🙏🏻.

In any case, from what you tell me, your App for WearOS is already able to do what I asked in the post, that is, create a communication between a SmartWatch/SmartBand and the mobile phone on which the Juggluco (reader) runs to read the data on the watch in some way, without third-party intermediaries (which entrust the operating system to slower and more fragile types of connections with uncovered areas and problems of various kinds, such as the "lag" of seconds or even problematic operation with the dial you mentioned), and this already comforts me.
I will definitely go (in the evening, because now I don't have much time) to better investigate the "watch" section from the settings in the menu on the left that you suggested to me to find out more.

In any case, however, I didn't understand one thing about the SmartBands and WearOS.
From what I understood preliminarily, the fastest method is Juggluco Wear (with WearOS), which directly shows the glucose value received from Juggluco (reader on the phone) also on the watch via Bluetooth, and therefore this suggests to me that it is the fastest method and free from particular problems (such as those mentioned with the server delay), but I have not understood well if it is possible to do this also on a SmartBand (Xiaomi MiBand for example) or is it only feasible on SmartWatches.

Can it be done on SmartBands?
Does WearOS also run on SmartBands (like Xiaomi MiBand)?

For me it would be enough to just be able to see the glucose value that updates, with the trend arrow and a small portion of the graph (2/3 hours before the real-time value) at most, then for the rest of the more advanced functions such as insulin doses and historical consultation of the graph for example, I would use Juggluco on the phone with more convenience.
As I was saying, I would like to avoid using systems that use servers and internet data connections, and particular bridges (which can have sudden downtime and potentially introduce excessive delays and disadvantages). And I would prefer to see my glucose in real time also on the watch (MiBand for example) as it happens on Juggluco on my cell phone.
I am considering buying the MiBand-8, because I see it as very well made and complete and also responsive, but I don't know if the recently released MiBand-9 is better and I'm still evaluating. 😇

**Comment by j-kaltes** [comment]
Xiaomi MiBand 8 runs a very slow CPU with very little RAM (500kb, https://en.wikipedia.org/wiki/Xiaomi_Mi_Band). WearOS needs 1000 times that much RAM. No way can you run WearOS on such a Smartband. WearOS is Android with some watch specific additions and a few parts removed. In practice you can only run WearOS only on a watch on which WearOS is already installed.
The main advantage of buying a WearOS watch is that Juggluco running on WearOS can directly connect with the sensor via Bluetooth, so that you don't need to take a phone with you. The main disadvantage is that the battery lasts 16 hours when you buy a cheap version. Very expensive WearOS watches can increase it to more then 24 hours.
When you have no problem with carrying a phone, you can also see a very recent glucose value on a Garmin watch running [Kerfstok](https://www.juggluco.nl/Kerfstok/) . But Garmins battery also doesn't last very long and the Garmin watches on which you can run Kerfstok are very expensive.
There are cheap watches that can be used with Gadgetbridge which battery lasts a long time. You need to find out if there is a watch face that displays the glucose value received via Gadgetbridge.

**Reply by SideTipeOne** [reply]
Thanks again for your time. 🙏🏻

I'm only answering you now because I've spent the last 2 days between commitments and work and trying to understand in the very little free time what "GadgetBridge" is (which I still haven't understood in its entirety, but I understood that it is a system that would replace the app that manages the SmartBand device and allows some functions). 😇
In any case I necessarily need to delve deeper into this issue of GadgetBridge to be more familiar. Unfortunately, despite having searched for some video tutorials online by some users that described a bit how it works, I didn't find much. So I need to search better.
In any case I read a post on Reddit (quite recent, from 6 months ago), in which a user asked for help because he couldn't connect his SmartBand ("MiBand-8") to GadgetBridge, but thanks to the help of a user he managed to solve the problem and also the author of the post confirmed that it works.
So this makes me think that the "MiBand-8" can communicate with GadgetBridge, so I hope it can also have a "watchface" capable of transmitting the glucose values ​​received from GB" (I have no reason to think that it is not possible with the "9" either, considering the fact that as I said I am evaluating it to have a product better updated to current standards). I am not sure but I will try to make sure of this, even if unfortunately the information available online is very fragmented and also not very understandable to a completely novice user like me.

I do not know how the watchface that receives the values ​​from the other systems that we have excluded for their excessive compromises (such as Xdrip etc..) works on a SmartBand, but I believe (and hope) that it can also be replicated with GadgetBridge. The thing that makes me a little suspicious and that makes me doubtful is that from what I understand the MiBand must be disconnected from its native App to work with GadgetBridge, and I do not know if this disassociation brings with it limits on the possible customization of watchfaces.
I don't understand anything about it, I'm really illiterate on the technical level in this regard, but I hope it can be done.
I am a member of a Facebook group of Diabetic users who use these systems, so Juggluco and also watches (among which very few with SmartBands), but almost all use systems like Xdrip and similar. I don't know if this is motivated by the fact that there is some intrinsic and insurmountable technical limit to the use of systems like GadgetBridge, or if it can be an unawareness of this possibility.
